File: Z_Deep_Learning_SGH/zaj_3/regresja_logistyczna.R
regresja_logistyczna <- function(x,y){
   X=as.matrix(cbind(matrix(1, nrow(x), 1),x))   
   w<- matrix(0, ncol(X), 1)
   w_old=w+10E9
   while ( norm(w-w_old, type="2")>0.00001 ){
	w_old=w
	p=exp(X%*%w)/(1+exp(X%*%w))
	V=diag(c(p*(1-p)))
	w=w+solve(t(X)%*%V%*%X,t(X)%*%(y-p))
   }
   rownames(w)<-NULL
   return(w)
}

File: Z_Deep_Learning_SGH/zaj_3/regresja_logistyczna.m
function w=regresja_logistyczna(x,y)
X=[ones(size(x,1),1),x]; 
w=rand(size(X,2),1)/1000;

w_old=w+inf;
while norm(w-w_old)>0.00001
    w_old=w;
    p=exp(X*w)./(1+exp(X*w));
    V=diag(p.*(1-p));
    w=w+(X'*V*X)\X'*(y-p);
end





